promptolution/optimizers/opro.py

```diff
@@ -0,0 +1,60 @@
+from typing import List
+
+import numpy as np
+
+from promptolution.llms.base_llm import BaseLLM
+from promptolution.optimizers.base_optimizer import BaseOptimizer
+
+
+class Opro(BaseOptimizer):
+    """
+    Opro: Optimization by PROmpting
+
```

Comment:
lets describe the idea at least high level s.t. users does not have to skim the paper if not familiar with the base idea